json_schema/1.0.0.schema.json

```diff
@@ -0,0 +1,558 @@
+          "format": "uri"
+        }
+      }
+    },
+    "relatedProject": {
+      "type": "object",
+      "description": "A related project.",
+      "required": [
+        "id"
+      ],
+      "properties": {
+        "src": {
+          "type": "string",
+          "description": "An external source where more metadata about this dependency can be found.",
+          "format": "uri"
+        },
+        "id": {
+          "type": "string",
+          "description": "A unique ID for this Project."
+        }
+      }
+    },
+    "dependency": {
+      "allOf": [{ "$ref": "#/definitions/relatedProject"}],
+      "required": [
+        "id",
+        "version"
+      ],
+      "properties": {
+        "version": {
+          "oneOf": [
+            {
+              "type": "array",
+              "description": "An array listing compatible versions.",
+              "items": {
+                "type": "string",
+                "description": "A compatible version"
+              }
+            },
+            {
+              "type": "string",
+              "description": "A SemVer comparison string.",
+              "examples": [
+                ">= 25.0.219"
+              ]
+            }
+          ]
+        },
+        "allowed": {
+          "description": "Whether or not this dependency is allowed. Used with conditions.",
+          "$ref": "#/definitions/condition"
+        },
+        "required": {
+          "description": "Whether or not this dependency is required.",
+          "oneOf": [
+            {
+              "type": "boolean"
+            },
+            {
+              "$ref": "#/definitions/condition"
+            }
+          ]
+        }
+      }
+    },
+    "condition": {
+      "type": "object",
+      "description": "Condition match",
+      "additionalProperties": false,
+      "properties": {
+        "environment": {
+          "type": "object",
+          "description": "Minecraft Game Environment",
+          "additionalProperties": false,
+          "properties": {
+            "client": {
+              "type": "boolean",
+              "description": "Client-side"
+            },
+            "server": {
+              "type": "boolean",
+              "description": "Server-side"
+            }
+          }
+        }
+      }
+    },
+    "installation": {
+      "type": "object",
+      "description": "Installation information",
+      "additionalProperties": false,
+      "required": [
+        "method"
+      ],
+      "properties": {
+        "method": {
+          "type": "string",
+          "description": "The installation method to use.",
+          "enum": [
+            "placeInDirectory",
+            "versionJsonInstall",
+            "runInstaller",
+            "classpathLibrary",
+            "instanceInstall",
+            "jarMod",
+            "other"
+          ]
+        },
+        "directory": true,
+        "versionJson": true
```

Comment:
According to the [JSON schema specification](https://json-schema.org/draft/2019-09/json-schema-core.html#rfc.section.9.3.2.1) every value of the `properties` field must be a valid JSON schema. A boolean shouldn’t meet that requirement. I solved that with `dependencies`: https://github.com/expeehaa/modip/commit/92f95771cc42b6e7252e09c87352bfa31aea3e74

This also is a problem in lines 175, 176, 256, 266, 267.

Reply:
I have given your version a try, and unfortunately VSCode thinks valid syntax is invalid :/ I'm not sure if the issue is with the schema or with VSC. I have managed to fix the problem you brought up, but it may not be ideal.